src/test/file/test_file.c

```diff
@@ -259,6 +259,17 @@ static void _test_tmpfile() {
     assert_nonneg_errno(fclose(file));
 }
 
+static void _test_dup() {
+    g_auto(AutoDeleteFile) adf = _create_auto_file();
+    int fd, fd2;
+
+    assert_nonneg_errno(fd = open(adf.name, O_RDONLY));
+    assert_nonneg_errno(fd2 = dup(fd));
```

Comment:
Do we want to expand this test, to make sure that something we wrote to `fd` _before_ the dup is actually readable from `fd2` _after_ the dup?

Reply:
I added more to this test.